__tests__/commands/install/integration.js

```diff
@@ -212,6 +212,26 @@ test.concurrent('install file: protocol with relative paths', (): Promise<void>
   });
 });
 
+test.concurrent('install file: protocol without cache', async (): Promise<void> => {
+  const fixturesLoc = path.join(__dirname, '..', '..', 'fixtures', 'install');
+  const compLoc = path.join(fixturesLoc, 'install-file-without-cache', 'comp', 'index.js');
+
+  return fs.writeFile(compLoc,'foo\n').then(() =>
+    runInstall({noLockfile: true}, 'install-file-without-cache', async (config) => {
+      assert.equal(
+        await fs.readFile(path.join(config.cwd, 'node_modules', 'comp', 'index.js')),
+        'foo\n',
+      );
+      await fs.writeFile(compLoc, 'bar\n');
+      await runInstall({noLockfile: true}, 'install-file-without-cache');
+      assert.equal(
+        await fs.readFile(path.join(config.cwd, 'node_modules', 'comp', 'index.js')),
```

Comment:
I think here you still read from the first install.
config.cwd comes as argument to runInstall.
Instead make this check inside the runInstall callback and double check that we are not testing the same location.